### DISCUSSION TITLE
Can I send an image as a numpy array as query params in GET request ?

### DISCUSSION THREAD

**Discussion by yichong96** [fastapi/fastapi#4226] [Questions] [answered]
I want to do an image search service whereby the client sends an image to search for and the server returns top - k similar images. This seems like a GET request style pattern. Does FASTapi allow numpy arrays in the query parameters ? 

Or should I use a POST request instead and send the image in the body ? 

Thank you very much !

**Comment by simondale00** [comment, marked as answer]
Hey, @yichong96. I'd recommend using `POST` for this use-case. It's purpose built to accept binary data in the request body. Whereas with `GET` your clients will have to convert the binary image data to a custom format that's URL-encoded; that can be a barrier to entry to use the API.

That and there are some practical limits to URL length; Some browsers limits URL length. And the backend server access logs will capture the full URL by default in a log; you'll likely run into logging problems if each log line contains megabytes of data.

Best,
Simon
